src/Character/Controller/BaseController.lua

```diff
@@ -272,14 +271,15 @@ function BaseController:UpdateCharacter(): ()
         local LowerTorsoCFrame = HumanoidRootPartCFrame * self.Character.Attachments.HumanoidRootPart.RootRigAttachment.CFrame * self.Character.Motors.Root.Transform * self.Character.Attachments.LowerTorso.RootRigAttachment.CFrame:Inverse()
         local UpperTorsoCFrame = LowerTorsoCFrame * self.Character.Attachments.LowerTorso.WaistRigAttachment.CFrame * self.Character.Motors.Waist.Transform * self.Character.Attachments.UpperTorso.WaistRigAttachment.CFrame:Inverse()
         local HeadCFrame = UpperTorsoCFrame * self.Character.Attachments.UpperTorso.NeckRigAttachment.CFrame * self.Character.Motors.Neck.Transform * self.Character.Attachments.Head.NeckRigAttachment.CFrame:Inverse()
-        CameraService:UpdateCamera(HeadCFrame * self.Character.Head:GetEyesOffset())
-    else
-        --Update the camera based on the last CFrame if the motors can't update (not in Workspace).
-        local CurrentCameraCFrame = Workspace.CurrentCamera.CFrame
+		CameraService:UpdateCamera(HeadCFrame * self.Character.Head:GetEyesOffset())
+		self.LastCameraCFrame = Workspace.CurrentCamera.CFrame
+		self.LastHeadCFrame = VRHeadCFrame
+	else
+		--Update the camera based on the last CFrame if the motors can't update (not in Workspace).
+		local LastCameraCFrame = self.LastCameraCFrame or CFrame.new()
         local LastHeadCFrame = self.LastHeadCFrame or CFrame.new()
         local HeadCFrame = self:ScaleInput(VRInputService:GetVRInputs()[Enum.UserCFrame.Head])
-        Workspace.CurrentCamera.CFrame = CurrentCameraCFrame * LastHeadCFrame:Inverse() * HeadCFrame
-        self.LastHeadCFrame = HeadCFrame
+		Workspace.CurrentCamera.CFrame = LastCameraCFrame * LastHeadCFrame:Inverse() * HeadCFrame
```

Comment:
Pretty certain this will cause problems when `Workspace.CurrentCamera.HeadLocked` is `true`. Does `CameraService:UpdateCamera` work at all?